Lib/test/test_memoryview.py

```diff
@@ -71,6 +71,36 @@ def setitem(value):
         m = None
         self.assertEqual(sys.getrefcount(b), oldrefcount)
 
+    def test_setitem_readonly_and_writable_buffer(self):
+        if not self.rw_type:
+            self.skipTest("no writable type to test")
+        b = self.rw_type(self._source)
+        oldrefcount = sys.getrefcount(b)
+        m = self._view(b)
+        readonly_m = m.cast("B", readonly=True)
```

Comment:
I would test `m.cast(m.format, readonly=True)` instead, to make sure that only changing writability is supported.

Same for the other test below.